service-base/src/main/java/org/eclipse/hono/service/auth/device/X509AuthProvider.java

```diff
@@ -83,7 +84,12 @@ protected SubjectDnCredentials getCredentials(final JsonObject authInfo) {
             if (tenantId == null || subjectDn == null) {
                 return null;
             } else {
-                return SubjectDnCredentials.create(tenantId, subjectDn);
+                final JsonObject clientContext = new JsonObject();
+                final byte[] clientCert = authInfo.getBinary(CredentialsConstants.FIELD_CLIENT_CERT);
```

Comment:
IMHO this can throw a runtime exception if the property value is not a valid Base64 string ...